src/args.rs

```diff
@@ -49,17 +64,48 @@ impl Args {
         self.paths.is_empty()
     }
 
-    pub(crate) fn output_mode(&self) -> OutputMode {
```

Comment:
It clicked for me that if (as I'd think we would) we would expose some other output-related "knobs" like `ripgrep` does eg `--heading`, `--column` then an enumerated `OutputMode` is not the right abstraction

So refactored to get rid of `OutputMode` and do something much more similar to what `ripgrep` does (make the `Args` "smarter" and specifically know how to produce configured eg searchers/printers based on its contained settings)